app.js

```diff
@@ -205,7 +205,7 @@ api.on('connection', (spark) => {
         })
       }
 
-      Nodes.addBlock(stats.id, stats.block, (err) => {
+      Nodes.addBlock(stats.id, stats.block, (err, stats) => {
```

Comment:
i believe `stats` will shadow the outer scope variable `stats` as far as i can remember does this handler not any return value expect err. Therefore `stats` would be undefined and shadow the outer scoped stats (which contains the block informations)

Reply:
This is a temp fix for the NaN values we're seeing in propagation times. I'll rename the inner scope var